alphagov/pay-selfservice
# PP-335 Add pagination to transaction list

## WHAT

_A brief description of the pull request:_

 Previously all transactions ever were displayed by transaction list. Connector now returns transaction list as paginated collection. This is an integration with that work. This has not been done in full HATEOAS style as it was realised that to achieve the ACs, the application could not simply pass on the links from the connector, but would have to parse them in any case. It then seemed more sensible to implement pagination independently in selfservice, enabling flexibility should business requirements change. Added a paginator utility class where most of the logic is, and modified transaction_controller, connector_client and transaction_view to use this. Some tidying up and javascriptification of files has also been done.
## HOW

_Steps to test or reproduce:_
Correct pagination links should appear on transaction list page.
Check unit tests, integration tests and ui tests for pagination.
When going through changes may be wise to use split view - some re-organisations aren't clearly displayed in 'unified' view.
## WHO

_Who should review this pull request:_
